### PR TITLE
Use pgrep -f on Linux, fallback for non-linux, non-apple

### PR DESCRIPTION
**Pull request by tkoolen**
`pgrep -f` is necessary because regular `pgrep` only looks at the first 15 characters on Linux, i.e. `drake-visualize` instead of `drake-visualizer`.

Also add test.

### CODE REVIEW

File: src/DrakeVisualizer.jl [hunk 1]
```diff
@@ -115,7 +115,15 @@ function new_window()
...
 end
 
-any_open_windows() = success(spawn(`pgrep $drake_visualizer_executable_name`))
+function any_open_windows()
+    if is_apple()
+        return success(spawn(`pgrep $drake_visualizer_executable_name`))
+    elseif is_linux()
+        return success(spawn(`pgrep -f $drake_visualizer_executable_name`))
+    else
+        return false # TODO
```
Comment (rdeits) [inline src/DrakeVisualizer.jl]: Maybe a warning here?

File: src/DrakeVisualizer.jl [hunk 2]
```diff
@@ -115,7 +115,15 @@ function new_window()
     proc
 end
 
-any_open_windows() = success(spawn(`pgrep $drake_visualizer_executable_name`))
+function any_open_windows()
+    if is_apple()
```
Comment (rdeits) [inline src/DrakeVisualizer.jl]: You can put `@static` before the `if` to have this magically happen at compile-time

**Comment by rdeits** [comment]
Thanks!

**Comment by tkoolen** [comment]
Test isn't passing on Travis because drake-visualizer just quits when there is no X server. I think I'll just get rid of the `@test`, just to make sure that the code runs without errors.

**Comment by rdeits** [comment]
Ok, no problem.

**Comment by tkoolen** [comment]
Addressed comments.

**Comment by rdeits** [comment]
Perfect, thanks!

**Review by rdeits — APPROVED** [review]

**Comment by tkoolen** [comment]
Can't really get code coverage for the `else`.

**Pull request merged**